### PR TITLE
gh-101100: Fix Sphinx warnings in `whatsnew/2.1.rst`

### PR DESCRIPTION
**Pull request by hugovk**
Same rationale as https://github.com/python/cpython/pull/112351.

Fix these 29 Sphinx warnings:

```
Doc/whatsnew/2.1.rst:51: WARNING: py:func reference target not found: g
Doc/whatsnew/2.1.rst:105: WARNING: py:func reference target not found: g
Doc/whatsnew/2.1.rst:162: WARNING: py:meth reference target not found: __cmp__
Doc/whatsnew/2.1.rst:182: WARNING: py:meth reference target not found: __lt__
Doc/whatsnew/2.1.rst:184: WARNING: py:meth reference target not found: __le__
Doc/whatsnew/2.1.rst:186: WARNING: py:meth reference target not found: __gt__
Doc/whatsnew/2.1.rst:188: WARNING: py:meth reference target not found: __ge__
Doc/whatsnew/2.1.rst:190: WARNING: py:meth reference target not found: __eq__
Doc/whatsnew/2.1.rst:192: WARNING: py:meth reference target not found: __ne__
Doc/whatsnew/2.1.rst:207: WARNING: py:func reference target not found: cmp
Doc/whatsnew/2.1.rst:248: WARNING: py:mod reference target not found: regex
Doc/whatsnew/2.1.rst:263: WARNING: py:mod reference target not found: regex
Doc/whatsnew/2.1.rst:275: WARNING: py:mod reference target not found: regex
Doc/whatsnew/2.1.rst:368: WARNING: py:func reference target not found: f
Doc/whatsnew/2.1.rst:382: WARNING: py:func reference target not found: memoize
Doc/whatsnew/2.1.rst:400: WARNING: py:exc reference target not found: weakref.ReferenceError
Doc/whatsnew/2.1.rst:444: WARNING: py:class reference target not found: UserDict
Doc/whatsnew/2.1.rst:586: WARNING: py:mod reference target not found: inspect.py
Doc/whatsnew/2.1.rst:586: WARNING: py:mod reference target not found: pydoc.py
Doc/whatsnew/2.1.rst:586: WARNING: py:mod reference target not found: pydoc.py
Doc/whatsnew/2.1.rst:619: WARNING: py:class reference target not found: SequenceMatcher
Doc/whatsnew/2.1.rst:631: WARNING: py:mod reference target not found: minidom
Doc/whatsnew/2.1.rst:645: WARNING: py:func reference target not found: asctime
Doc/whatsnew/2.1.rst:645: WARNING: py:func reference target not found: localtime
Doc/whatsnew/2.1.rst:725: WARNING: py:meth reference target not found: xreadlines
Doc/whatsnew/2.1.rst:725: WARNING: py:func reference target not found: xrange
Doc/whatsnew/2.1.rst:725: WARNING: py:func reference target not found: xreadlines
Doc/whatsnew/2.1.rst:725: WARNING: py:meth reference target not found: readlines
Doc/whatsnew/2.1.rst:739: WARNING: py:meth reference target not found: popitem
```

* Issue: gh-101100

----
:books: Documentation preview :books:: https://cpython-previews--112357.org.readthedocs.build/

### CODE REVIEW

File: Doc/whatsnew/2.1.rst [hunk 1]
```diff
@@ -175,21 +175,21 @@ In Python 2.1, rich comparisons were added in order to support this need.
...
-| ``>=``    | :meth:`__ge__` |
-+-----------+----------------+
-| ``==``    | :meth:`__eq__` |
-+-----------+----------------+
-| ``!=``    | :meth:`__ne__` |
-+-----------+----------------+
++-----------+------------------------+
+| Operation | Method name            |
++===========+========================+
+| ``<``     | :func:`~object.__lt__` |
```
Comment (terryjreedy) [inline Doc/whatsnew/2.1.rst]: This looks like a mistake.  Copy/paste error?
```suggestion
| ``<``     | :meth:`~object.__lt__` |
```

File: Doc/whatsnew/2.1.rst [hunk 2]
```diff
@@ -584,11 +584,11 @@ available from the Distutils SIG at https://www.python.org/community/sigs/curren
 New and Improved Modules
 ========================
 
-* Ka-Ping Yee contributed two new modules: :mod:`inspect.py`, a module for
-  getting information about live Python code, and :mod:`pydoc.py`, a module for
+* Ka-Ping Yee contributed two new modules: :mod:`!inspect.py`, a module for
+  getting information about live Python code, and :mod:`!1pydoc.py`, a module for
```
Comment (terryjreedy) [inline Doc/whatsnew/2.1.rst]: Extra char '1' added?
```suggestion
  getting information about live Python code, and :mod:`!pydoc.py`, a module for
```

**Review by terryjreedy — CHANGES_REQUESTED** [review]
Possibly 2 changes needed.

**Comment by hugovk** [comment]
I have made the requested changes; please review again.

**Review by terryjreedy — APPROVED** [review]

**Pull request merged**